### PR TITLE
Authentication System Added

### PR DESCRIPTION
**Pull request by Pratham-Mishra04**
Joi Validator added with all the changes specified by the Sr. Core

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,131 +1 @@
...
-
-
-
-<!-- MARKDOWN LINKS & IMAGES -->
-<!-- https://www.markdownguide.org/basic-syntax/#reference-style-links -->
-[csivitu-shield]: https://img.shields.io/badge/csivitu-csivitu-blue
-[csivitu-url]: https://csivit.com
-[issues-shield]: https://img.shields.io/github/issues/csivitu/Template.svg?style=flat-square
-[issues-url]: https://github.com/csivitu/Template/issues
+# Riddler-Backend-2022
```
Comment (sanjaybaskaran01) [inline README.md:1]: Removes existing README.MD Template

File: app.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import express from "express";
+import morgan from "morgan"
+import dotenv from 'dotenv'
+import path from 'path'
+import AppError from "./utils/AppError.js";
+import { noURL } from "./Controllers/ErrorController.js";
```
Comment (sanjaybaskaran01) [inline app.js:6]: Causes error since controller folder is not capital but here it is capital
image

File: package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "jsonwebtoken": "^8.5.1",
+    "mongoose": "^6.4.0",
+    "morgan": "^1.10.0",
+    "multer": "^1.4.5-lts.1",
+    "nodemailer": "^6.7.7",
+    "nodemon": "^2.0.18",
+    "sharp": "^0.30.7",
+    "slugify": "^1.6.5",
+    "validator": "^13.7.0"
+  }
```
Comment (sanjaybaskaran01) [inline package.json:30]: libphonenumber-js package is not here, so it will raise error

File: routers/userRouter.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import express from "express";
+import { signup, login, protect, logout } from "../Controllers/authController.js";
```
Comment (sanjaybaskaran01) [inline routers/userRouter.js:2]: Import URL is incorrect because of capitalization

File: routers/userRouter.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import { getAllUsers, UpdatePassword, getUser, updateUser, deleteUser, filterBody, forgotPassword, resetPassword, uploadProficPic, resizePic } from "../Controllers/userController.js";
```
Comment (sanjaybaskaran01) [inline routers/userRouter.js:3]: Import URL is incorrect because of capitalization

File: utils/joiValidator.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import Joi from 'joi'
+import User from '../models/userModel.js';
+import catchAsync from './catchAsync.js'
+
+const joiCreateSchema = Joi.object({
+    name:Joi.string().pattern(/^[A-Za-z]+$/, 'alpha').required(),
+    email:Joi.string().email().lowercase().custom(async (value, helper)=>{
+        const user= await User.find({email: value});
```
Comment (sanjaybaskaran01) [inline utils/joiValidator.js:9]: Interesting implementation :boom:

**Review by sanjaybaskaran01 — CHANGES_REQUESTED** [review]
Create envHandler so that it throws error if necessary env is not present, Make changes to current reviews so that it is executable without errors

File: utils/joiValidator.js (hunk 6 above)
Comment (Pratham-Mishra04) [reply utils/joiValidator.js:9]: Thank You😁

**Review by sanjaybaskaran01 — APPROVED** [review]

**Review by sanjaybaskaran01 — APPROVED** [review]

**Pull request merged**